src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,111 @@
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { Filter } from '../../types/Filter';
+
+type Props = {
+  todos: Todo[],
+  filter: (filter: Filter) => void
+  removeCompletedTodos: () => void,
+};
+
+export const Footer: React.FC<Props> = ({
+  todos,
+  filter,
+  removeCompletedTodos: removeAll,
+}) => {
+  const [activeFilter, setActiveFilter] = useState({
+    all: true,
+    active: false,
+    completed: false,
+  });
+
+  const loadFilteringTodos = (isActive: boolean) => {
+    if (!isActive) {
+      filter('active');
+      setActiveFilter({
+        all: false,
+        active: true,
+        completed: false,
+      });
+    } else {
+      filter('completed');
+      setActiveFilter({
+        all: false,
+        active: false,
+        completed: true,
+      });
+    }
+  };
+
+  const allTodo = () => {
+    filter('all');
+    setActiveFilter({
+      all: true,
+      active: false,
+      completed: false,
+    });
+  };
+
+  const isCompletedTodos = todos.some(todo => todo.completed);
+  const todosLeft = todos.length - todos.filter(item => item.completed).length;
+  const removeCompletedTodos = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    removeAll();
+
+    return null;
+  };
+
+  return (
+    <footer className="todoapp__footer">
+      <span className="todo-count">
+        {`${todosLeft} items left`}
+      </span>
+
+      <nav className="filter">
+        <a
+          href="#/"
+          className={
+            classNames('filter__link', { selected: activeFilter.all })
```

Comment:
No need to use `activeFilter` here. You can check the class the next way:
```
selected: currentFiter === 'all'
```
The same is for other class name ('active', 'completed')